### DISCUSSION TITLE
iterate all keys of the input

### DISCUSSION THREAD

**Discussion by setop** [mitsuhiko/minijinja#523] [Q&A] [answered]
Let's say my input is the following:

```json
{
  "a": 1,
  "f": 6,
  "m": 14
}
```

But I don't know the keys in advance.

I'd like to iterate over keys in a template like this one:

```j2
{% for k, v in input.items() %}
{# do something with key #}
{# do something with value #}
{% endfor %}
```

How do I get this list of keys at the root of the data ?

**Comment by mitsuhiko** [comment, marked as answer]
Out of the box you cannot, but you can use a self-referntial context and register a name for the context itself. There is an example for this here: https://github.com/mitsuhiko/minijinja/blob/main/examples/self-referential-context/src/main.rs

In that example the name `CONTEXT` is reserved for the context itself.
